tests/orm/nodes/data/test_xy.py

```diff
@@ -51,3 +51,21 @@ def test_constructor():
     node = XyData(x_array, y_array, x_name='x_name', x_units='x_unit', y_names='y_name', y_units='y_units')
     assert numpy.array_equal(node.get_x()[1], x_array)
     assert numpy.array_equal(node.get_y()[0][1], y_array)
+
+
+def test_get_y_arraynames():
+    """Test retrieving y array names."""
+    x_array = numpy.array([1, 2])
+    y_array1 = numpy.array([3, 4])
+    y_array2 = numpy.array([5, 6])
+
+    node = XyData()
+    node.set_x(x_array, 'x_name', 'x_unit')
+    node.set_y([y_array1, y_array2], ['y_name1', 'y_name2'], ['y_unit1', 'y_unit2'])
```

Comment:
Since we are here, to further improve to make the behavior act as expected, can you add tests that when the shape of y arguments passed in is not consistent? I assume it can have more data validation in `set_y` to make sure when setting the y data user always get proper panic if the data passed in is in correct.